web/src/components/MemoContent/MermaidBlock.tsx

```diff
@@ -1,17 +1,35 @@
 import { useEffect, useRef } from "react";
+import { getSystemColorScheme } from "@/helpers/utils";
+import { useCommonContext } from "@/layouts/CommonContextProvider";
 
 interface Props {
   content: string;
 }
 
 const MermaidBlock: React.FC<Props> = ({ content }: Props) => {
   const mermaidDockBlock = useRef<null>(null);
+  const commonContext = useCommonContext();
 
+  const handleMermaidTheme = () => {
+    if (commonContext.appearance === "dark") {
+      return "dark";
+    } else if (commonContext.appearance === "light") {
+      return "neutral";
+    } else {
+      if (getSystemColorScheme() === "dark") {
+        return "dark";
+      } else if (getSystemColorScheme() === "light") {
+        return "neutral";
+      } else {
+        return "default";
+      }
+    }
+  };
```

Comment:
It should be possible to simplify the theme selection logic to this. 

```suggestion
const { mode } = useColorScheme();

const handleMermaidTheme = () => {
  return mode == 'dark'
    ? 'dark'
    : 'default'
};
```

`app.tsx` checks `commonContext.appearance` and `getSystemColorScheme()` to determine the theme and uses `setMode(currentAppearance);` to set `mui-joy`'s `mode` var with that value so no need to redo it.

Note:
`mui`s `mode` is the only variable out of the vars that are set for theme that get dynamically updates after page load. You can see this in the browser's local storage view, where a copy of this values is stored, as the OS/browser theme changes. 
I was going to suggest adding `mode` to the `useEffect` variables to track so that any OS/browser changes take effect as the user makes the change, but it doesn't work. `Mermaid.js` doesnt allow for `re-initialization` out of the box and some workarounds are needed to get it to happen. see [here](https://github.com/mermaid-js/mermaid/issues/1945). I dont think is worth the effort if a user can refresh to see the new theme and the user potentially doesn't ever change their theme once its set.